### PR TITLE
Feature/backend/app connection

### PR DESCRIPTION
**Pull request by successsb89**
# Description

_Briefly explain what the PR does._
Mainly focused on appointment module connection with some UI update.

list of related files
- AddAppointmentDialog.tsx
- AppointmentDropdown.tsx
- dropConfirmationDialog.js

---

# Related Jira Issue (ID/URL)
RTTER-127(https://rtterror.atlassian.net/browse/RTTER-127)
RTTER-417(https://rtterror.atlassian.net/browse/RTTER-417)

# Related GitHub Issues

- connection of the module appointment(https://github.com/kevinlizh1992/rtterror/pull/54)

# Checklist

- [x] Code compiles correctly
- [x] Code is covered by current tests or new tests
- [x] Code is formatted correctly
- [x] Extended the README / documentation, if necessary

---

# Comment/Discussion
-connect backend data with appointment form.
-appointment scheduler is updated based on in charged  employee.
image
-update confirmation details
image

### CODE REVIEW

File: backend/src/config/index.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default {
       port: process.env.MYSQL_PORT || '3306',
       database: process.env.MYSQL_DATABASE || 'database1',
       username: process.env.MYSQL_USERNAME || 'root',
-      password: process.env.MYSQL_PASSWORD || 'password',
+      password: process.env.MYSQL_PASSWORD || '',
```
Comment (yuxiang-zhang) [inline backend/src/config/index.ts:16]: Remove this change, as from previous experience, this causes issues for automated tests

File: frontend/pages/appointment/helper.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+const generateSchedules = (appointments) => {
```
Comment (yuxiang-zhang) [inline frontend/pages/appointment/helper.js:1]: NextJS expects the directory `pages/` to only contain page components. You need to move this file elsewhere.

**Review by yuxiang-zhang — CHANGES_REQUESTED** [review]
- [ ] All changes related to `yarn.lock` `package-lock.json` need to be removed. The automated tests are setup with yarn only. 
- [ ] Remove every line of `console.log`; all these logs will overflow the console. 
- [ ] Remove all commented codes, they are code smells.

File: backend/src/config/index.ts (hunk 1 above)
Comment (successsb89) [reply backend/src/config/index.ts:16]: what is the meaning of removing the change?

Comment (yuxiang-zhang) [reply backend/src/config/index.ts:16]: The default `password` is expected by the automated tests.

**Review by yuxiang-zhang — CHANGES_REQUESTED** [review]
- [ ] `yarn.lock` and `backend/yarn.lock` should be restored. 
- [ ] major code smells exist from sonar cloud analysis
- [ ] e2e tests are failing

**Comment by successsb89** [comment]
resolve all the issues. please review it @yuxiang-zhang 
- contains all correct files.
- code smalls decreased by 1.
- passed e2e test.

**Review by yuxiang-zhang — APPROVED** [review]
- All code smells are resolved.
- Visual checks have been performed. 
- E2e tests passed.

**Review by kevinlizh1992 — APPROVED** [review]

**Pull request merged**